### PR TITLE
Refactor auth

### PR DESCRIPTION
**Pull request by jmhooper**
This PR refactors the authentication logic to use a simpler, more vanilla passport configuration. The most complicated component of the authentication setup now is the "verify callback" for the GitHub authentication strategy, which is how it should be.

This PR simplifies things by abandoning a lot of the generated code which stood up username / password authentication which we do not use. This was abandoned for the GitHub auth strategy with a verify callback that checks that the user is on one of the whitelisted GitHub teams.

Finally, this PR adds tests to confirm that authentication works as expected.

### CODE REVIEW

**Comment by brandocalrissian** [comment]
git conflict! i think it's a simple resolution...

**Comment by jmhooper** [comment]
Just rebased!

File: test/api/requests/auth.test.js [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+        })
+        .reply(401)
+
+      request("http://localhost:1337")
+        .get("/auth/github/callback?code=invalid-code")
+        .expect(401, done)
+    })
+
+    it("should respond with a 401 if the user is not in a whitelisted GitHub organization", done => {
+      githubAPINocks.githubAuth("unatuhorized-user", [{ id: 654321 }])
```
Comment (jmhooper) [inline test/api/requests/auth.test.js]: *unauthorized

**Comment by brandocalrissian** [comment]
reviewed with help of @jmhooper. good to go. minor comments.

**Pull request merged**